diff --git a/base/loading.jl b/base/loading.jl
index f63bb6bfef..87c05000b1 100644
--- a/base/loading.jl
+++ b/base/loading.jl
@@ -1491,7 +1491,7 @@ function stale_cachefile(modpath::String, cachefile::String)
                 # Issue #13606: compensate for Docker images rounding mtimes
                 # Issue #20837: compensate for GlusterFS truncating mtimes to microseconds
                 ftime = mtime(f)
-                if ftime != ftime_req && ftime != floor(ftime_req) && ftime != trunc(ftime_req, digits=6)
+                if ftime != ftime_req && ftime != floor(ftime_req) && ftime != trunc(ftime_req, digits=6) && ftime != 1.0 && ftime != 315532800.0
                     @debug "Rejecting stale cache file $cachefile (mtime $ftime_req) because file $f (mtime $ftime) has changed"
                     return true
                 end
